### PR TITLE
Large variant of Media object pattern

### LINKED ISSUE #1315: Bigger variant of Media Pattern
# Pattern proposal/amendment

## Is this a proposal or an amendment?

- Amendment

If amendment, what is the name of the pattern to be amended?

Media object pattern (bigger version)

## Visual

**Large screen**

image

**Small screen**

image

**Sizing**

image

## Context

Newly designed Snap Details page includes a component consisting of a snap icon, name, publisher, install button etc.

This seems very similar in layout to current Media Object pattern but uses different sizing (icon is bigger, larger heading).

It would be nice to have such variant in code Vanilla. Currently to achieve similar look we need to use grid directly which behaves differently on smaller screens (stacks on top of each other).

Context: https://github.com/CanonicalLtd/snapcraft-design/issues/54

### PR DESCRIPTION
**Pull request by bartaz**
## Done

As part of implementing Snap details page we needed bigger variant of media pattern (as mentioned in #1315) that was implemented in https://github.com/canonical-websites/snapcraft-flask/pull/49

Here it is ported to base Vanilla.

It adds new modifier to `.p-media-object` class: `.p-media-object--large`. It has:
- that is a larger version of image 96x96px
- bigger title (defaults to heading 1 styles)
- larger spacing between paragraphs

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- Go to 'Media object - large' example: http://0.0.0.0:8101/vanilla-framework/examples/patterns/media-object/media-object-large/

## Details

Fixes  #1315

## Screenshots

screen shot 2017-09-14 at 15 06 47

### CODE REVIEW

File: scss/_patterns_media-object.scss [hunk 1]
```diff
@@ -12,6 +12,13 @@
 }
 
 @mixin vf-p-media-object {
+  %media-image {
+    border-radius: 4px;
+    height: 3.75rem;
+    margin-right: 1rem;
```
Comment (barrymcgee) [inline scss/_patterns_media-object.scss]: Can you use [spacing variables](https://github.com/vanilla-framework/vanilla-framework/blob/develop/scss/_settings_spacing.scss) here, please?

**Comment by anthonydillon** [comment]
As this is a block or pattern level modification. I think the modification should just be on the block and the elements within should respond accordingly.

**Comment by bartaz** [comment]
@anthonydillon @barrymcgee Thanks for comments.

I updated PR to use spacing variables and moved modifier to block level class name `.p-media-object--large`.

**Comment by bartaz** [comment]
@anthonydillon Sure. I just updated media object docs to include example of new variant.

**Comment by kwm14** [comment]
## Visual review

Referencing @spencerbygraves design [spec](https://user-images.githubusercontent.com/9105999/29959607-0b1fd8bc-8ef0-11e7-90a8-cf3e17e956a4.png)

- [ ] Update element margin to: `2.5rem`
- [ ] Update content margin-top to: `0.75rem`

**Comment by kwm14** [comment]
LGTM 👍

**Comment by bartaz** [comment]
@kwm14 @barrymcgee 
I'm not sure if including top/bottom padding was intended for this pattern.

Yes, on snap details page media object has padding around it, but also it has bottom border - which makes it a good candidate to be used inside of `p-strip is-bordered` - and that's how it was implemented.

Now, with additional padding the spacing is doubled (both from media-object and strip patterns).
Removing strip will make bottom border disappear. Unless there is a possibility of making p-strip with border but without padding?

**Comment by kwm14** [comment]
@bartaz when reviewing your PR I synced up with @spencerbygraves and referenced his [spec](https://user-images.githubusercontent.com/9105999/29959607-0b1fd8bc-8ef0-11e7-90a8-cf3e17e956a4.png), so I was just following his guidelines when adding my design +1 to make sure the pattern matched what he had designed.

If this was a particular use-case to include top/bottom padding for this component I'm happy for you to remove and roll back to your build which had `1rem` padding top and bottom?

**Comment by bartaz** [comment]
@kwm14 Thanks for comment.

I believe that the spec was meant to define spacing for snap details page, which in here we mistakenly interpreted as defining element margins for the pattern itself.

Large media pattern on it's own can keep default spacing defined by Vanilla and this will allow to use it inside of p-strip without adding unexpected spacing. I'll propose a bugfix for that.

**Comment by kwm14** [comment]
@bartaz ah ok thanks for clarifying that and thanks for filing a bug. I'll keep an eye on it as we have a few components that need some design tweaks so I'll make a plan for component updates.

**Comment by bartaz** [comment]
Fix proposed in #1352

**Pull request merged**